doc/source/whatsnew/v1.0.2.rst

```diff
@@ -23,6 +23,7 @@ Fixed regressions
 - Fixed regression where :func:`read_pickle` raised a ``UnicodeDecodeError`` when reading a py27 pickle with :class:`MultiIndex` column (:issue:`31988`).
 - Fixed regression in :class:`DataFrame` arithmetic operations with mis-matched columns (:issue:`31623`)
 - Fixed regression in :meth:`GroupBy.agg` calling a user-provided function an extra time on an empty input (:issue:`31760`)
+- Fixed regression in :meth:`pandas.core.internals.blocks.Block.setitem` throwing an error when a datetime64[ns, tz] value is provided (:issue:`32395`)
```

Comment:
What's the user-facing bug thats fixed? i.e. `Series.__seitem___` of `loc.__setitem__` or ...?  users shouldnt need to know about Block

Reply:
It's `loc.__setitem__`. Should I change to `:meth:loc.__setitem__`?